From: Michael Froman <mfroman@mozilla.com>
Date: Tue, 30 Dec 2025 12:01:47 -0600
Subject: (tmp-cherry-pick) Revert "Expose ParseIceServers in api/"
 (a4867cda19)

This reverts commit dbfb5fb5d50de4abd71924d8a7390ece28c270a1.

Reason for revert: Seems order of RTC_EXPORT and [[nodiscard]] breaks bots.

Original change's description:
> Expose ParseIceServers in api/
>
> Per discussion on crrev.com/c/6972479
>
> Bug: chromium:443019066
> Change-Id: Id02bc85dd55a246faf036bdbe8918c6ad6e2089b
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/413080
> Reviewed-by: Guido Urdaneta <guidou@webrtc.org>
> Reviewed-by: Harald Alvestrand <hta@webrtc.org>
> Auto-Submit: Tony Herre <herre@google.com>
> Commit-Queue: Guido Urdaneta <guidou@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#45775}

Bug: chromium:443019066
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Change-Id: Ic634e5bb5cdfe6f8fde0c361e928c37de8d06f74
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/413300
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: Harald Alvestrand <hta@webrtc.org>
Commit-Queue: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#45783}
---
 api/BUILD.gn              | 16 ----------------
 api/DEPS                  |  5 -----
 api/ice_server_parsing.cc | 30 ------------------------------
 api/ice_server_parsing.h  | 35 -----------------------------------
 4 files changed, 86 deletions(-)
 delete mode 100644 api/ice_server_parsing.cc
 delete mode 100644 api/ice_server_parsing.h

diff --git a/api/BUILD.gn b/api/BUILD.gn
index 218c9b1328..b48a9aa46f 100644
--- a/api/BUILD.gn
+++ b/api/BUILD.gn
@@ -1804,19 +1804,3 @@ rtc_library("frame_transformer_factory") {
     "video:video_frame_metadata",
   ]
 }
-
-rtc_library("ice_server_parsing") {
-  visibility = [ "*" ]
-  sources = [
-    "ice_server_parsing.cc",
-    "ice_server_parsing.h",
-  ]
-  deps = [
-    ":libjingle_peerconnection_api",
-    ":rtc_error",
-    "../p2p:port_allocator",
-    "../pc:ice_server_parsing",
-    "../rtc_base:socket_address",
-    "../rtc_base/system:rtc_export",
-  ]
-}
diff --git a/api/DEPS b/api/DEPS
index 558cc3bb3d..51421a4e23 100644
--- a/api/DEPS
+++ b/api/DEPS
@@ -111,11 +111,6 @@ specific_include_rules = {
     "+modules/include/module_fec_types.h",
   ],
 
-  "ice_server_parsing\.h": [
-    "+p2p/base/port_allocator.h",
-    "+rtc_base/socket_address.h",
-  ],
-
   "jsep\.h": [
     "+absl/strings/has_absl_stringify.h",
     "+absl/strings/str_format.h",
diff --git a/api/ice_server_parsing.cc b/api/ice_server_parsing.cc
deleted file mode 100644
index b33ddbc64e..0000000000
--- a/api/ice_server_parsing.cc
+++ /dev/null
@@ -1,30 +0,0 @@
-/*
- *  Copyright 2025 The WebRTC project authors. All Rights Reserved.
- *
- *  Use of this source code is governed by a BSD-style license
- *  that can be found in the LICENSE file in the root of the source
- *  tree. An additional intellectual property rights grant can be found
- *  in the file PATENTS.  All contributing project authors may
- *  be found in the AUTHORS file in the root of the source tree.
- */
-
-#include "api/ice_server_parsing.h"
-
-#include <set>
-#include <vector>
-
-#include "api/peer_connection_interface.h"
-#include "api/rtc_error.h"
-#include "p2p/base/port_allocator.h"
-#include "pc/ice_server_parsing.h"
-#include "rtc_base/socket_address.h"
-
-namespace webrtc {
-
-RTCError ParseIceServers(const PeerConnectionInterface::IceServers& servers,
-                         std::set<SocketAddress>* stun_servers,
-                         std::vector<RelayServerConfig>* turn_servers) {
-  return ParseIceServersOrError(servers, stun_servers, turn_servers);
-}
-
-}  // namespace webrtc
diff --git a/api/ice_server_parsing.h b/api/ice_server_parsing.h
deleted file mode 100644
index 3b20394e01..0000000000
--- a/api/ice_server_parsing.h
+++ /dev/null
@@ -1,35 +0,0 @@
-/*
- *  Copyright 2025 The WebRTC project authors. All Rights Reserved.
- *
- *  Use of this source code is governed by a BSD-style license
- *  that can be found in the LICENSE file in the root of the source
- *  tree. An additional intellectual property rights grant can be found
- *  in the file PATENTS.  All contributing project authors may
- *  be found in the AUTHORS file in the root of the source tree.
- */
-
-#ifndef API_ICE_SERVER_PARSING_H_
-#define API_ICE_SERVER_PARSING_H_
-
-#include <set>
-#include <vector>
-
-#include "api/peer_connection_interface.h"
-#include "api/rtc_error.h"
-#include "p2p/base/port_allocator.h"
-#include "rtc_base/socket_address.h"
-#include "rtc_base/system/rtc_export.h"
-
-namespace webrtc {
-
-// Parses the URLs for each server in `servers` to build `stun_servers` and
-// `turn_servers`. Can return SYNTAX_ERROR if the URL is malformed, or
-// INVALID_PARAMETER if a TURN server is missing `username` or `password`.
-RTC_EXPORT [[nodiscard]] RTCError ParseIceServers(
-    const PeerConnectionInterface::IceServers& servers,
-    std::set<SocketAddress>* stun_servers,
-    std::vector<RelayServerConfig>* turn_servers);
-
-}  // namespace webrtc
-
-#endif  // API_ICE_SERVER_PARSING_H_
